mono/metadata/exception.c

```diff
@@ -1435,13 +1429,14 @@ mono_error_set_simple_file_not_found (MonoError *error, const char *file_name, g
 void
 mono_error_set_argument_out_of_range (MonoError *error, const char *name)
 {
-	//FIXMEcoop
-	mono_error_set_exception_instance (error, mono_get_exception_argument_out_of_range (name));
+	ERROR_DECL (error_creating_exception);
+	mono_error_set_exception_handle (error, mono_new_exception_argument_out_of_range (name, error_creating_exception));
+	mono_error_cleanup (error_creating_exception);
 }
 
 MonoExceptionHandle
 mono_error_convert_to_exception_handle (MonoError *error)
 {
-	//FIXMEcoop mono_error_convert_to_exception is raw pointer
-	return MONO_HANDLE_NEW (MonoException, mono_error_convert_to_exception (error));
+	return mono_error_ok (error) ? MONO_HANDLE_CAST (MonoException, NULL_HANDLE)
```

Comment:
This "optimization" is actually critical. Otherwise we were using lots of handles.